emb_tau_id_sfs_ul.sh

```diff
@@ -0,0 +1,666 @@
+echo "##############################################################################################"
+echo "#      Producing shapes for ${CHANNEL}-${ERA}-${NTUPLETAG}                                         #"
+echo "##############################################################################################"
+
+# if the output folder does not exist, create it
+if [ ! -d "$shapes_output" ]; then
+    mkdir -p $shapes_output
+    mkdir -p "${shapes_output}_mm"
+fi
+
+echo "${shapes_output}_mm"
+
+if [[ $MODE == "CONTROL" ]]; then
+    source utils/setup_root.sh
+    python shapes/produce_shapes.py --channels $CHANNEL \
+        --directory $NTUPLES \
+        --${CHANNEL}-friend-directory $XSEC_FRIENDS \
+        --era $ERA --num-processes 3 --num-threads 9 \
+        --vs-jet-wp $WP \
+        --vs-ele-wp ${VS_ELE_WP} \
+        --optimization-level 1 --skip-systematic-variations \
+        --special-analysis "TauID" \
+        --control-plot-set ${VARIABLES} \
+        --output-file $shapes_output  --xrootd  --validation-tag $TAG 
+fi
+
+if [[ $MODE == "CONTROLREGION" ]]; then
+    source utils/setup_root.sh
+    python shapes/produce_shapes.py --channels mm \
+        --directory $NTUPLES \
+        --mm-friend-directory $XSEC_FRIENDS \
+        --era $ERA --num-processes 3 --num-threads 9 \
+        --vs-jet-wp $WP \
+        --vs-ele-wp ${VS_ELE_WP} \
+        --optimization-level 1 --skip-systematic-variations \
+        --special-analysis "TauID" \
+        --output-file "${shapes_output}_mm"  --xrootd --validation-tag $TAG
+fi
+
+PROCESSES="emb"
+number="_emb_ssos"
+if [[ $MODE == "LOCAL" ]]; then
+    source utils/setup_root.sh
+    python shapes/produce_shapes.py --channels $CHANNEL \
+        --directory $NTUPLES \
+        --${CHANNEL}-friend-directory $XSEC_FRIENDS \
+        --era $ERA --num-processes 3 --num-threads 9 \
+        --vs-jet-wp $WP \
+        --vs-ele-wp ${VS_ELE_WP} \
+        --optimization-level 1 \
+        --special-analysis "TauID" \
+        --process-selection $PROCESSES \
+        --control-plot-set ${VARIABLES} \
+        --optimization-level 1 \
+        --output-file $shapes_output$number --xrootd --validation-tag $TAG --es
+fi
+
+if [[ $MODE == "CONDOR" ]]; then
+    source utils/setup_root.sh
+    echo "[INFO] Running on Condor"
+    echo "[INFO] Condor output folder: ${CONDOR_OUTPUT}"
+    bash submit/submit_shape_production_ul.sh $ERA $CHANNEL \
+        "singlegraph" $TAG 0 $NTUPLETAG $CONDOR_OUTPUT "TauID" 0 $WP $VS_ELE_WP
+    echo "[INFO] Jobs submitted"
+fi
+
+if [[ $MODE == "MERGE" ]]; then
+    source utils/setup_root.sh
+    echo "[INFO] Merging outputs located in ${CONDOR_OUTPUT}"
+    hadd -j 5 -n 600 -f $shapes_rootfile ${CONDOR_OUTPUT}/../analysis_unit_graphs-${ERA}-${CHANNEL}-${NTUPLETAG}-${TAG}/*.root
+fi
+
+if [[ "${ERA}" == "2018"  ||  "${ERA}" == "2017" ]]; then 
+    datacard_era=${ERA}
+elif [[ "${ERA}" == "2016postVFP"  ||  "${ERA}" == "2016preVFP" ]]; then
+   datacard_era="2016"
```

Comment:
we do not destinguish between 2016 pre and post ?

Reply:
We do and effectively we  get different scale factors for different eras, however morphing script doesn't differentiate between eras